testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -11692,9 +11683,9 @@ void assertSubFunReturns(boolean binary, String s, int start,
       f12.checkScalar("NVL2(NULL, 3.0, 2.111)", "2.111", "DECIMAL(4, 3) NOT NULL");
       f12.checkScalar("NVL2(3.111, 3.1415926, 2.111)", "3.1415926", "DECIMAL(8, 7) NOT NULL");
 
-      f12.checkNullValueWithNullableType("nvl2('ab', CAST(NULL AS VARCHAR(6)), 'def')");
-      f12.checkNullValueWithNullableType("nvl2(NULL, 'abc', NULL)");
-      f12.checkNullValueWithNullableType("nvl2(NULL, NULL, NULL)");
+      f12.checkNull("nvl2('ab', CAST(NULL AS VARCHAR(6)), 'def')");
```

Comment:
I thought you also meant to change checkNull to assert that the type is nullable.
Or is that already happening?

Reply:
This PR makes `checknull` to support the check nullable type.